File: src/types/attachment.ts
export type AttachmentType = 'file' | 'link';

export interface Attachment {
  id: string;
  session_id: string;
  type: AttachmentType;
  label?: string;
  file_path?: string;
  url?: string;
  created_at: string;
}

export interface CreateAttachmentInput {
  session_id: string;
  type: AttachmentType;
  label?: string;
  file_path?: string;
  url?: string;
}

File: src-tauri/src/models/attachment.rs
use serde::{Deserialize, Serialize};

#[derive(Debug, Clone, Serialize, Deserialize, PartialEq)]
#[serde(rename_all = "snake_case")]
pub enum AttachmentType {
    File,
    Link,
}

#[derive(Debug, Clone, Serialize, Deserialize)]
pub struct Attachment {
    pub id: String,
    pub session_id: String,
    #[serde(rename = "type")]
    pub type_: AttachmentType,
    pub label: Option<String>,
    pub file_path: Option<String>,
    pub url: Option<String>,
    pub created_at: String,
}

#[derive(Debug, Clone, Serialize, Deserialize)]
pub struct CreateAttachmentInput {
    pub session_id: String,
    #[serde(rename = "type")]
    pub type_: AttachmentType,
    pub label: Option<String>,
    pub file_path: Option<String>,
    pub url: Option<String>,
}
